### DISCUSSION TITLE
Anomaly Detection Use Case

### DISCUSSION THREAD

**Discussion by 9race** [amazon-science/chronos-forecasting#164] [Q&A]
Hi all, 

Thanks for this open source library. I was wondering if there's any way to perform anomaly detection with Chronos?

Thanks!

**Comment by abdulfatir** [comment]
@9race you can potentially use Chronos for forecasting-based anomaly detection, but we haven't tried it.

**Comment by AccsoMaximilianSaerov** [comment]
My team and I are currently trying to find a way to do so. Our idea was either to predict the delta t between 2 anomalies, i.e. transform the dataset into a list of delta t values, or to transform the dataset into a "binary" classification, where all non-anomalies are zero and all anomalies are non-zero. This tends to work pretty okayish when using a dataset with a lot of anomalies and a limited amount of data points overall. We used a synthetic sine curve where we set all values >0.9/>-0.9 as an anomaly. Unfortunately as the maximum recommended prediction length is 64, it is very unlikely that an anomaly is present in there. 

Do you have any idea how we could proceed here?

**Reply by abdulfatir** [reply]
@mxsrv We just released Chronos-2. It supports wider quantile grid from 0.01 to 0.99 and long prediction horizons (up to 1024 in a single forward pass). This should address your concerns. Feel free to continue the discussion in #335.
